types/transform-pouch/index.d.ts

```diff
@@ -0,0 +1,21 @@
+// Type definitions for transform-pouch 1.0
+// Project: https://github.com/pouchdb-community/transform-pouch
+// Definitions by: Lucas Rainett <https://github.com/lucasrainett>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="pouchdb-core" />
+
+declare namespace PouchDB {
+    interface Database<Content extends {} = {}> {
+        transform<NewContent>(config: {
+            incoming?(doc: Core.Document<Content>): Core.Document<NewContent> | Promise<Core.Document<NewContent>>;
+            outgoing?(doc: Core.Document<NewContent>): Core.Document<Content> | Promise<Core.Document<Content>>;
+        }): void;
+    }
```

Comment:
```suggestion
    interface Database<Content extends {} = {}> {
        transform<NewContent>(config: {
            incoming?(doc: Core.Document<Content>): Core.Document<NewContent> | Promise<Core.Document<NewContent>>;
            outgoing?(doc: Core.Document<NewContent>): Core.Document<Content> | Promise<Core.Document<Content>>;
        }): void;
        // api.filter provided for backwards compat with the old "filter-pouch"
        filter<NewContent>(config: {
            incoming?(doc: Core.Document<Content>): Core.Document<NewContent> | Promise<Core.Document<NewContent>>;
            outgoing?(doc: Core.Document<NewContent>): Core.Document<Content> | Promise<Core.Document<Content>>;
        }): void;
    }
```
Should package support `filter` alias?
https://github.com/pouchdb-community/transform-pouch/blob/master/index.js#L26
Sorry for duplicated code, not sure how to properly alias method in augmentation (not sure if that is possible either)

Reply:
Hi, We can, the docs only mention about the filter as:

`
Note: This plugin was formerly known as filter-pouch, but was renamed to be less confusing. The filter() API is still supported, but deprecated.
`

Apart from that the filter is not mentioned anywhere else.

I will send the update soon